rust/otap-dataflow/crates/otap/src/retry_processor.rs

```diff
@@ -225,53 +182,17 @@ impl RetryProcessorMetrics {
             SignalType::Traces => self.produced_items_traces_refused.add(n),
         }
     }
-    /// Increment produced.items with outcome=failure for the given signal by n
-    pub fn add_produced_failure(&mut self, st: SignalType, n: u64) {
-        match st {
-            SignalType::Logs => self.produced_items_logs_failure.add(n),
-            SignalType::Metrics => self.produced_items_metrics_failure.add(n),
-            SignalType::Traces => self.produced_items_traces_failure.add(n),
-        }
-    }
-}
-
-/// Configuration for the retry processor
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct RetryConfig {
-    /// Maximum number of retry attempts before dropping a message
-    pub max_retries: usize,
-    /// Initial delay in milliseconds before the first retry
-    pub initial_retry_delay_ms: u64,
-    /// Maximum delay in milliseconds between retries
-    pub max_retry_delay_ms: u64,
-    /// Multiplier applied to delay for exponential backoff
-    pub backoff_multiplier: f64,
-    /// Maximum number of messages that can be pending retry
-    pub max_pending_messages: usize,
-    /// Interval in seconds for cleanup of expired messages
-    pub cleanup_interval_secs: u64,
-}
 
-impl Default for RetryConfig {
-    fn default() -> Self {
-        Self {
-            max_retries: 3,
-            initial_retry_delay_ms: 1000,
-            max_retry_delay_ms: 30000,
-            backoff_multiplier: 2.0,
-            max_pending_messages: 10000,
-            cleanup_interval_secs: 60,
+    /// Increment retry_attempts for the given signal by 1.
+    pub fn add_retry_attempts(&mut self, st: SignalType) {
```

Comment:
Maybe use the term `increment` instead of `add` now that we only add `1` to the value? Or we could also just let it be to stay consistent with the other metric update method names.

```suggestion
    pub fn increment_retry_attempts(&mut self, st: SignalType) {
```